src/comm/TcpAcceptor.cc

```diff
@@ -250,17 +249,16 @@ void
 Comm::TcpAcceptor::logAcceptError(const ConnectionPointer &tcpClient) const
 {
     AccessLogEntry::Pointer al = new AccessLogEntry;
-    CodeContext::Reset(al);
     al->tcpClient = tcpClient;
     al->url = "error:accept-client-connection";
     al->setVirginUrlForMissingRequest(al->url);
```

Comment:
If there is no compelling reason to keep this ALE-detailing code outside the CallBack() enclosure, please move it there. It is the right thing to do from exception handling point of view -- any exception should be treated as specific to the transaction being logged (because all the manipulations in this code a specific to that transaction rather than transaction-agnostic TcpAcceptor code).

I realize that we will be switching to a "half-baked" ALE context if we move this detailing code. However, I think the point of the code context switching is not the availability of all the details. We would gladly use as many details as we can, but even if there are no details[^1] available, we still want to protect the surrounding TcpAcceptor context from being blamed for this transaction-specific failure.

[^1]: other than a unique ALE ID (not currently supported, IIRC, but will be eventually when ALE/MasterXaction objects are merged)